### PR TITLE
fix undef variable check puppet6

### PR DESCRIPTION
**Pull request by fe80**
Hello,

The variable check of some template not work with puppet6

### CODE REVIEW

File: templates/_auth_ldap.erb [hunk 1]
```diff
@@ -14,13 +14,13 @@ provider_url =
   providerUrl="<%= provider_url %>"
   authenticationMethod="simple"
   forceBindingLogin="<%= @auth_config['ldap']['force_binding'] %>"
-<%- if @auth_config['ldap']['force_binding_use_root'] != :undef -%>
+<%- if !@auth_config['ldap']['force_binding_use_root'].nil? && @auth_config['ldap']['force_binding_use_root'] != :undef -%>
```
Comment (alexjfisher) [inline templates/_auth_ldap.erb:17]: Considering `force_binding_use_root` [defaults to `false`](https://github.com/voxpupuli/puppet-rundeck/blob/2fdbe3e6999809249659620258897497f4276c4d/manifests/params.pp#L185) and not `undef`, I'm not convinced this conditional is needed at all.

**Comment by alexjfisher** [comment]
Thanks for the PR!  I think I've come across a similar fix in another module before. So we can confirm everything now works correctly (and won't be broken in future), would you be able to write any tests?

File: templates/_auth_ldap.erb (hunk 1 above)
Comment (alexjfisher) [reply templates/_auth_ldap.erb]: @fe80 I've taken the liberty of pushing a commit onto your branch (something you _can_ turn off when you open a PR) that removes this conditional and adds a test for this ldap property.  Hoping you could follow the pattern and add tests for the other properties you've fixed?  Feel free to discard/squash it if you want to add more tests or write them in a different way. :)

**Comment by fe80** [comment]
I've had more test and fix a bug (I've miss a `.nil?` condition)

**Review by bastelfreak — APPROVED** [review]

**Comment by kenyon** [comment]
Obsoleted by #520.

**Pull request closed without merging**